src/DotLiquid/Context.cs

```diff
@@ -422,29 +433,42 @@ private object Resolve(string key, bool notifyNotFound = true)
         /// the hierarchy
         /// </summary>
         /// <param name="key"></param>
+        /// <param name="variable"></param>
         /// <returns></returns>
-        private object FindVariable(string key)
+        private bool TryFindVariable(string key, out object variable)
         {
+            bool foundVariable = false;
             Hash scope = Scopes.FirstOrDefault(s => s.ContainsKey(key));
-            object variable = null;
+            object ret = null;

```

Comment:
I like that you eliminated the mutability of the previous local by having a different variable for the non-liquidized version, but what would be a more descriptive name than 'ret'?

Reply:
To me that is short for return - a reserved keyword. Would you prefer foundValue?

Reply:
It is not the return here because both the return type now is the boolean for found or not found, and the out variable needs to be liquidize before been assigned.
As to `foundValue`, does value vs variable carry the nuance that one is liquidized and not the other?